src/Illuminate/Support/Collection.php

```diff
@@ -1088,6 +1088,22 @@ public function split($numberOfGroups)
     }
 
     /**
+     * Split a collection based on condition into groups.
+     *
+     * @param  closure  $callback
+     * @return static
+     */
+    public function splitBy($callback)
+    {
+        $results = $this->reduce(function ($results, $row) use ($callback) {
+            $index = $callback($row) ? 0 : 1;
+            $results[$index][] = $row;
+            return $results;
+         }, []);
+        return new static($results);
```

Comment:
@lucasmichot I also feel strange when I didn't see it running, It's working now.

Thanks